### PR TITLE
Cortx-28282-Refactoring-Error-Messages

### PR DESCRIPTION
**Pull request by AshwiniS-Borse**
Signed-off-by: AshwiniS-Borse <ashwini.borse@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

Collection is green
# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by AshwiniS-Borse** [comment]
@apoorva-rao @VikasKumarC Please review

**Comment by AshwiniS-Borse** [comment]
https://jts.seagate.com/browse/TEST-38968 - run with code changes
https://jts.seagate.com/browse/TEST-37801 - with main branch code

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]

File: tests/s3/test_bucket_policy.py [hunk 1]
```diff
@@ -262,15 +263,14 @@ def delete_bucket_policy_with_err_msg(self,
         try:
             s3_bkt_policy_obj_two.delete_bucket_policy(bucket_name)
```
[the remark this answers is not in the document]
Comment (AshwiniS-Borse) [reply tests/s3/test_bucket_policy.py:264]: Agree.
This change is required to be made in the whole code base and needs validation, can't be done in 1 day.
@apoorva-rao @seagate-sarang-sawant Can we have 1 task for this ?

File: tests/s3/test_bucket_policy.py [hunk 2]
```diff
@@ -262,15 +263,14 @@ def delete_bucket_policy_with_err_msg(self,
...
+            assert errconst.ACCESS_DENIED_ERR_KEY in error.message, error.message
         resp = acl_obj_one.put_bucket_acl(
             bucket_name, acl="private")
         assert resp[0], resp[1]
         resp = s3_obj_one.delete_bucket(bucket_name, force=True)
         assert resp[0], resp[1]
-        self.log.info(
-            "Delete bucket policy should through error message %s",
-            test_config["error_message"])
+        self.log.info("Delete bucket policy should through error message %s",
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_policy.py:272]: It would be better to revert the changes to the indentation in this call.

File: tests/s3/test_bucket_location.py [hunk 3]
```diff
@@ -125,24 +126,22 @@ def test_get_bkt_loc_valid_bkt_272(self):
     @CTFailOn(error_handler)
     def test_get_bkt_loc_bkt_not_present_273(self):
         """verify get bucket location for the bucket which is not present."""
-        self.log.info(
-            "Verify get bucket location for the bucket which is not present")
-        self.log.info(
-            "Step 1 : Check the bucket location on non existing bucket %s ",
+        self.log.info("STARTED: Verify get bucket location for the bucket which is "
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_location.py:129]: Indentation of the original code needs to be restored here for lines 129-131

File: tests/s3/test_account_user_management.py [hunk 4]
```diff
@@ -626,12 +622,9 @@ def test_create_user_with_existing_name_2081(self):
...
         except CTException as error:
             self.log.debug(error.message)
-            assert_in(
-                "EntityAlreadyExists",
-                error.message,
-                error.message)
-        self.log.info(
-            "Could not create user with existing name %s", str(self.user_name))
+            assert_in(errconst.DUPLICATE_USER_ERR_KEY, error.message, error.message)
+        self.log.info("Could not create user with existing name %s",
```
Comment (apoorva-rao) [inline tests/s3/test_account_user_management.py:626]: Indentation of the earlier code for self.log.info call needs to be restored

File: tests/s3/test_bucket_policy.py [hunk 5]
```diff
@@ -1422,18 +1422,15 @@ def test_560(self):
...
             self.s3_bkt_policy_obj.delete_bucket_policy(self.bucket_name)
         except CTException as error:
-            assert err_msg in error.message, error.message
-        self.log.info(
-            "Step 1: Delete bucket policy should through error message %s",
-            err_msg)
-        self.log.info(
-            "ENDED: Apply Delete-bucket-policy on non existing bucket.")
+            assert errconst.NO_BUCKET_OBJ_ERR_KEY in error.message, error.message
+        self.log.info("Step 1: Delete bucket policy should through error message %s",
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_policy.py:1431]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 6]
```diff
@@ -158,7 +159,8 @@ def test_2434(self):
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.info(error)
-            assert_utils.assert_in("NoSuchTagSetError", str(error.message), error.message)
+            assert_utils.assert_in(errconst.BKT_SET_TAG_ERR,
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:162]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 7]
```diff
@@ -227,10 +229,11 @@ def test_2436(self):
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.info(error)
-            assert_utils.assert_in("InvalidTagError", str(error.message), error.message)
+            assert_utils.assert_in(errconst.BKT_INVALID_TAG_ERR,
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:232]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 8]
```diff
@@ -239,8 +242,8 @@ def test_2436(self):
     @CTFailOn(error_handler)
     def test_2437(self):
         """Create a tag having values up to 256 Unicode characters in length."""
-        self.log.info(
-            "STARTED: Create a tag having values up to 256 Unicode characters in length")
+        self.log.info("STARTED: Create a tag having values up to 256 Unicode "
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:245]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 9]
```diff
@@ -306,10 +309,11 @@ def test_2438(self):
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.info(error)
-            assert_utils.assert_in("InvalidTagError", str(error.message), error.message)
+            assert_utils.assert_in(errconst.BKT_INVALID_TAG_ERR,
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:312]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 10]
```diff
@@ -306,10 +309,11 @@ def test_2438(self):
...
+                    str(error.message), error.message)
         self.log.info("Step 2: Setting tag for a bucket failed with %s", self.bucket_name)
-        self.log.info(
-            "ENDED: Create a tag having values more than 512 Unicode characters in length")
+        self.log.info("ENDED: Create a tag having values more than 512 Unicode characters "
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:315]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 11]
```diff
@@ -366,7 +370,8 @@ def test_2440(self):
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.info(error)
-            assert_utils.assert_in("InvalidTagError", str(error.message), error.message)
+            assert_utils.assert_in(errconst.BKT_INVALID_TAG_ERR,
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:373]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 12]
```diff
@@ -483,7 +488,8 @@ def test_2444(self):
                 assert_utils.assert_false(resp[0], resp[1])
             except CTException as error:
                 self.log.info(error)
-                assert_utils.assert_in("InvalidTagError", str(error.message), error.message)
+                assert_utils.assert_in(errconst.BKT_INVALID_TAG_ERR,
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:491]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 13]
```diff
@@ -514,7 +520,8 @@ def test_2445(self):
...
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:523]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 14]
```diff
@@ -542,9 +549,9 @@ def test_2446(self):
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.info(error)
-            assert_utils.assert_in("MalformedXML", str(error.message), error.message)
-        self.log.info(
-            "Step 2: Setting bucket tags with duplicate keys failed with %s",
+            assert_utils.assert_in(errconst.BKT_TAG_DUPLICATE_KEY_ERR,
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:552]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 15]
```diff
@@ -542,9 +549,9 @@ def test_2446(self):
...
+                                str(error.message), error.message)
+        self.log.info("Step 2: Setting bucket tags with duplicate keys failed with %s",
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:554]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 16]
```diff
@@ -670,7 +677,8 @@ def test_2449(self):
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.info(error)
-            assert_utils.assert_in("NoSuchBucket", str(error.message), error.message)
+            assert_utils.assert_in(errconst.NO_BUCKET_OBJ_ERR_KEY,
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:680]: Indentation needs to be corrected

File: tests/s3/test_bucket_tagging.py [hunk 17]
```diff
@@ -679,7 +687,8 @@ def test_2449(self):
...
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:690]: Indentation needs to be corrected.

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
@AshwiniS-Borse I am cutting short my review after checking the first 10 files. Please fix the indentation across all 34 files (wherever applicable) to conform to the coding style we follow (https://google.github.io/styleguide/pyguide.html for reference, as mentioned in repo's wiki)

I have also made some suggestions for renaming some of the constants. Please check that as well.

File: tests/s3/test_bucket_location.py (hunk 3 above)
Comment (AshwiniS-Borse) [reply tests/s3/test_bucket_location.py:129]: In past review, I got review comment to restore such kind of indentation. I am unsure what to follow exactly.

File: tests/s3/test_bucket_location.py [hunk 18]
```diff
@@ -136,13 +137,12 @@ def test_get_bkt_loc_bkt_not_present_273(self):
...
             self.log.info(error)
-            assert_utils.assert_in("NoSuchBucket", str(
-                error.message), error.message)
-        self.log.info(
-            "Step 1 : Get bucket location on non existing bucket failed with error %s",
-            "NoSuchBucket")
-        self.log.info(
-            "Verify get bucket location for the bucket which is not present")
+            assert_utils.assert_in(errconst.NO_BUCKET_OBJ_ERR_KEY,
+                str(error.message), error.message)
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_location.py:141]: Indentation needs to be aligned, here is an example -
```suggestion
            assert_utils.assert_in(errconst.NO_BUCKET_OBJ_ERR_KEY,
                                   str(error.message), error.message)
```
There are other options as well, like what was original present at lines 141-145

This needs to be done in the block from 140 - 145

File: tests/s3/test_bucket_location.py [hunk 19]
```diff
@@ -272,10 +272,9 @@ def test_cross_account_get_bkt_loc_275(self):
             s3_obj_2.bucket_location(
                 self.bucket_name)
         except CTException as error:
-            assert_utils.assert_in("AccessDenied", str(
-                error.message), error.message)
-        self.log.info(
-            "Step 3 : Get bucket location with another account is failed"
+            assert_utils.assert_in(errconst.ACCESS_DENIED_ERR_KEY,
+                str(error.message), error.message)
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_location.py:276]: Please align this correctly or fix the indentation for lines 275 and 276

File: tests/s3/test_bucket_tagging.py [hunk 20]
```diff
@@ -169,8 +170,8 @@ def test_2434(self):
     @CTFailOn(error_handler)
     def test_2435(self):
         """Create a tag whose key is up to 128 Unicode characters in length."""
-        self.log.info(
-            "STARTED: Create a tag whose key is up to 128 Unicode characters in length")
+        self.log.info("STARTED: Create a tag whose key is up to 128 Unicode "
+            "characters in length")
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:174]: Please align this or fix indentation for lines 173 and 174

File: tests/s3/test_bucket_workflow_operations.py [hunk 21]
```diff
@@ -289,11 +289,10 @@ def test_name_formatting_1982(self):
...
             self.log.error(error.message)
-            assert "InvalidBucketName" in error.message, error.message
-        self.log.info(
-            "Creating a bucket with an IP address format is failed")
-        self.log.info(
-            "ENDED: Bucket names must not be formatted as an IP address (for example, 192.168.5.4)")
+            assert errconst.BKT_INVALID_NAME_ERR in error.message, error.message
+        self.log.info("Creating a bucket with an IP address format is failed")
+        self.log.info("ENDED: Bucket names must not be formatted as an IP address "
+            "(for example, 192.168.5.4)")
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_workflow_operations.py:295]: Please align this or fix the indentation for 294 and 295

File: tests/s3/test_bucket_workflow_operations.py [hunk 22]
```diff
@@ -799,31 +795,27 @@ def test_delete_multiobjects_432(self):
...
                 self.bucket_name, obj_lst)
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.error(error.message)
-            assert "NoSuchBucket" in error.message, error.message
-            self.log.info(
-                "Step 1: objects delete operation failed with error %s",
+            assert errconst.NO_BUCKET_OBJ_ERR_KEY in error.message, error.message
+            self.log.info("Step 1: objects delete operation failed with error %s",
                 "NoSuchBucket")
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_workflow_operations.py:809]: Please align this or fix the indentation for 808 and 809

File: tests/s3/test_bucket_workflow_operations.py [hunk 23]
```diff
@@ -799,31 +795,27 @@ def test_delete_multiobjects_432(self):
...
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.error(error.message)
-            assert "NoSuchBucket" in error.message, error.message
-            self.log.info(
-                "Step 2: List objects for non-existing bucket failed with error %s",
-                "NoSuchBucket")
+            assert errconst.NO_BUCKET_OBJ_ERR_KEY in error.message, error.message
+            self.log.info("Step 2: List objects for non-existing bucket failed with "
+                "error %s", "NoSuchBucket")
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_workflow_operations.py:818]: Please align this correctly or fix the indentation for 817 and 818

File: tests/s3/test_bucket_workflow_operations.py [hunk 24]
```diff
@@ -883,9 +873,8 @@ def test_delete_object_without_permission_434(self):
             assert_utils.assert_false(resp[0], res[1])
         except CTException as error:
             self.log.error(error.message)
-            assert "AccessDenied" in error.message, error.message
-            self.log.info(
-                "Step 3: deleting objects using account 2 failed with error %s",
+            assert errconst.ACCESS_DENIED_ERR_KEY in error.message, error.message
+            self.log.info("Step 3: deleting objects using account 2 failed with error %s",
                 "AccessDenied")
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_workflow_operations.py:878]: Please align this correctly or fix the indentation for 876 and 877

File: tests/s3/test_delete_account_temp_cred.py [hunk 25]
```diff
@@ -201,13 +202,10 @@ def test_4519(self):
...
             LOGGER.error(error.message)
-            assert_in(
-                test_4519_cfg["err_message"],
-                error.message,
-                error.message)
-        LOGGER.info(
-            "Step 5: Deleting account using invalid credentials failed with %s",
-            test_4519_cfg["err_message"])
+            assert_in(errconst.INVALID_ACCESSKEY_ERR_KEY,
+                error.message, error.message)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:206]: Please align this correctly or fix the indentation for 205 and 206

File: tests/s3/test_delete_account_temp_cred.py [hunk 26]
```diff
@@ -201,13 +202,10 @@ def test_4519(self):
...
+        LOGGER.info("Step 5: Deleting account using invalid credentials failed with %s",
+            errconst.INVALID_ACCESSKEY_ERR_KEY)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:208]: Please align this correctly or fix the indentation for 207 and 208

File: tests/s3/test_delete_account_temp_cred.py [hunk 27]
```diff
@@ -230,15 +226,11 @@ def test_4520(self):
...
-                error.message)
-        LOGGER.info(
-            "Step 4: Deleting non existing account with temp credentials failed with %s",
-            test_4520_cfg["err_message"])
-        LOGGER.info(
-            "ENDED: Delete non existing account with temp credentials.")
+            assert_in(errconst.INVALID_ACCESSKEY_ERR_KEY,
+                error.message, error.message)
+        LOGGER.info("Step 4: Deleting non existing account with temp credentials failed with %s",
+            errconst.INVALID_ACCESSKEY_ERR_KEY)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:232]: Please align this correctly or fix the indentation for 231 and 232

File: tests/s3/test_delete_account_temp_cred.py [hunk 28]
```diff
@@ -250,17 +242,16 @@ def test_4521(self):
...
         self.get_temp_creds(
             self.account_name,
             self.cfg["password"],
             duration=test_4521_cfg["time_duration"])
         time.sleep(test_4521_cfg["time_duration"])
-        LOGGER.info(
-            "Step 5: Deleting account %s using temporary credentials after expiry of time limit",
-            self.account_name)
+        LOGGER.info("Step 5: Deleting account %s using temporary credentials after "
+            "expiry of time limit", self.account_name)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:254]: Please align this correctly or fix the indentation for 253 and 254

File: tests/s3/test_delete_account_temp_cred.py [hunk 29]
```diff
@@ -269,16 +260,11 @@ def test_4521(self):
...
-                error.message)
-        LOGGER.info(
-            "Step 5: Deleting account using temporary credentials after"
-            " expiry of time limit failed with %s",
-            test_4521_cfg["err_message"])
-        LOGGER.info(
-            "ENDED: Delete account after 20 mins using temp credentials with expire time limit.")
+            assert_in(errconst.CRED_EXPIRE_ERR, error.message, error.message)
+        LOGGER.info("Step 5: Deleting account using temporary credentials after"
+            " expiry of time limit failed with %s", errconst.CRED_EXPIRE_ERR)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:265]: Please align this correctly or fix the indentation for 264 and 265

File: tests/s3/test_delete_account_temp_cred.py [hunk 30]
```diff
@@ -269,16 +260,11 @@ def test_4521(self):
...
+        LOGGER.info("ENDED: Delete account after 20 mins using temp credentials with expire "
+            "time limit.")
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:267]: Please align this correctly or fix the indentation for 266 and 267

File: tests/s3/test_delete_account_temp_cred.py [hunk 31]
```diff
@@ -287,48 +273,38 @@ def test_4521(self):
...
-                error.message)
-        LOGGER.info(
-            "Step 7: Creating a bucket using temp credentials of a deleted account failed with %s",
-            test_4522_cfg["err_message"])
-        LOGGER.info(
-            "ENDED: Perform S3 operations using deleted account temp credentials")
+            assert_in(errconst.INVALID_ACCESSKEY_ERR_KEY,
+                    error.message, error.message)
+        LOGGER.info("Step 7: Creating a bucket using temp credentials of a deleted account "
+            "failed with %s", errconst.INVALID_ACCESSKEY_ERR_KEY)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:306]: Please align this correctly or fix the indentation for 305 and 306

File: tests/s3/test_delete_account_temp_cred.py [hunk 32]
```diff
@@ -337,33 +313,27 @@ def test_4522(self):
     @CTFailOn(error_handler)
     def test_4523(self):
         """Delete account using temp cred where that account recently got deleted."""
-        LOGGER.info(
-            "STARTED: Delete account using temp cred where that account recently got deleted")
-        test_4523_cfg = S3_TMP_CRED_CFG["test_4523"]
+        LOGGER.info("STARTED: Delete account using temp cred where that account recently "
+            "got deleted")
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:317]: Please align this correctly or fix the indentation for 316 and 317

File: tests/s3/test_delete_account_temp_cred.py [hunk 33]
```diff
@@ -437,15 +396,12 @@ def test_4526(self):
...
-                error.message)
-        LOGGER.info(
-            "Step 4: Retrieving temporary credentials for the account failed with %s",
-            test_4526_cfg["err_message"])
-        LOGGER.info(
-            "ENDED: Delete account after 1 hour using temp credentials with expire time limit.")
+            assert_in(errconst.MAX_DUR_EXCEED_ERR,
+                    error.message, error.message)
+        LOGGER.info("Step 4: Retrieving temporary credentials for the account failed with %s",
+            errconst.MAX_DUR_EXCEED_ERR)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:402]: Please align this correctly or fix the indentation for 401 and 402

File: tests/s3/test_delete_account_temp_cred.py [hunk 34]
```diff
@@ -437,15 +396,12 @@ def test_4526(self):
...
-                test_4526_cfg["err_message"],
-                error.message,
-                error.message)
-        LOGGER.info(
-            "Step 4: Retrieving temporary credentials for the account failed with %s",
-            test_4526_cfg["err_message"])
-        LOGGER.info(
-            "ENDED: Delete account after 1 hour using temp credentials with expire time limit.")
+            assert_in(errconst.MAX_DUR_EXCEED_ERR,
+                    error.message, error.message)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:400]: Please align this correctly or fix the indentation for 399 and 400

File: tests/s3/test_delete_account_temp_cred.py [hunk 35]
```diff
@@ -437,15 +396,12 @@ def test_4526(self):
...
+        LOGGER.info("Step 4: Retrieving temporary credentials for the account failed with %s",
+            errconst.MAX_DUR_EXCEED_ERR)
+        LOGGER.info("ENDED: Delete account after 1 hour using temp credentials with expire "
+            "time limit.")
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:404]: Please align this correctly or fix the indentation for 403 and 404

File: tests/s3/test_delete_account_temp_cred.py [hunk 36]
```diff
@@ -478,13 +432,9 @@ def test_4692(self):
...
-                error.message)
-        LOGGER.info(
-            "Step 6: Deleting an account forcefully using temp credentials failed with %s",
-            test_4692_cfg["err_message"])
-        LOGGER.info(
-            "ENDED: Delete account forcefully using temp credentials "
+            assert_in(errconst.ACC_NOT_EMPTY_ERR,
+                    error.message, error.message)
+        LOGGER.info("Step 6: Deleting an account forcefully using temp credentials failed "
+            "with %s", errconst.ACC_NOT_EMPTY_ERR)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:438]: Please align this correctly or fix the indentation for 437 and 438

File: tests/s3/test_delete_account_temp_cred.py [hunk 37]
```diff
@@ -478,13 +432,9 @@ def test_4692(self):
...
-                test_4692_cfg["err_message"],
-                error.message,
-                error.message)
-        LOGGER.info(
-            "Step 6: Deleting an account forcefully using temp credentials failed with %s",
-            test_4692_cfg["err_message"])
-        LOGGER.info(
-            "ENDED: Delete account forcefully using temp credentials "
+            assert_in(errconst.ACC_NOT_EMPTY_ERR,
+                    error.message, error.message)
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:436]: Please align this correctly or fix the indentation for 435 and 436

File: tests/s3/test_delete_account_temp_cred.py [hunk 38]
```diff
@@ -478,13 +432,9 @@ def test_4692(self):
...
+        LOGGER.info("Step 6: Deleting an account forcefully using temp credentials failed "
+            "with %s", errconst.ACC_NOT_EMPTY_ERR)
+        LOGGER.info("ENDED: Delete account forcefully using temp credentials "
             "where that account contains some Resource")
```
Comment (apoorva-rao) [inline tests/s3/test_delete_account_temp_cred.py:440]: Please align this correctly or fix the indentation for 439 and 440

File: tests/s3/test_multipart_abort_copy.py [hunk 39]
```diff
@@ -379,7 +380,9 @@ def test_copy_multipart_upload_parallel_delete_29166(self):
                 self.s3_test_obj.object_info(bucket, self.object_name)
             except CTException as error:
                 self.log.error(error)
-                assert_utils.assert_equal(mp_config["error_msg"], error.message, error.message)
+                assert_s3_err_msg(errconst.RGW_HEAD_OBJ_ERR,
+                                errconst.CORTX_HEAD_OBJ_ERR,
+                                CMN_CFG["s3_engine"], error)
```
Comment (apoorva-rao) [inline tests/s3/test_multipart_abort_copy.py:385]: Please align this correctly or fix the indentation for 383, 384 and 385

File: tests/s3/test_multipart_upload.py [hunk 40]
```diff
@@ -576,15 +569,13 @@ def test_multipart_upload_all_parts_less_than_5mb_2296(self):
             assert_utils.assert_false(resp[0], resp[1])
         except CTException as error:
             self.log.error(error.message)
-            assert_utils.assert_in(
-                err_msg,
-                error.message,
-                error.message)
+            assert_utils.assert_in(errconst.MULTIPART_LIST_PART_LESS_ERR,
+                error.message, error.message)
```
Comment (apoorva-rao) [inline tests/s3/test_multipart_upload.py:573]: Please align this correctly or fix the indentation for 572 and 573

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
Added inline comments for indentation changes for first 17 files, please refactor all 34 accordingly.

File: commons/error_constants.py [hunk 41]
```diff
@@ -0,0 +1,104 @@
...
+NO_BUCKET_OBJ_ERR_KEY = "NoSuchBucket"
+NO_SUCH_KEY_ERR = "NoSuchKey"
+NO_SUCH_UPLOAD_ERR = "NoSuchUpload"
+NO_BUCKET_NAME_ERR = "Required parameter bucket_name not set"
+NOT_FOUND_ERR = "Not Found"
+
+#S3 Bucket Tagging
+BKT_SET_TAG_ERR = "NoSuchTagSetError"
+BKT_INVALID_TAG_ERR = "InvalidTagError"
+BKT_TAG_DUPLICATE_KEY_ERR = "MalformedXML"
```
Comment (archanabpatil) [inline commons/error_constants.py:72]: Can we name it MALFORMED_XML ,so name will specify what is error string

File: commons/error_constants.py [hunk 42]
```diff
@@ -0,0 +1,104 @@
...
+BKT_POLICY_INVALID_PRINCIPAL_ERR = "Invalid principal in policy"
+BKT_POLICY_MISSING_FIELD_ERR = "Missing required field"
+BKT_POLICY_UNKNOWN_FIELD_ERR = "Unknown field"
+BKT_POLICY_INVALID_ACTION_ERR = "invalid action"
+BKT_POLICY_EMPTY_ACTION_ERR = "Action cannot be empty"
+BKT_POLICY_INVALID_JSON_ERR = "This policy contains invalid Json"
+BKT_POLICY_NO_SUCH_ERR = "NoSuchBucketPolicy"
+
+#S3 versioning head obj error
+S3_VERSION_NOT_FOUND_GET_OBJ = "The specified version does not exist"
```
Comment (archanabpatil) [inline commons/error_constants.py:101]: line 101 and 102 are same errors  , but different var names ?

**Review by archanabpatil — CHANGES_REQUESTED** [review]

File: commons/error_constants.py (hunk 41 above)
Comment (AshwiniS-Borse) [reply commons/error_constants.py:72]: This error is related to bucket tag operation and hence named accordingly

File: commons/error_constants.py (hunk 42 above)
Comment (AshwiniS-Borse) [reply commons/error_constants.py:101]: removed one constant

**Review by archanabpatil — APPROVED** [review]

File: commons/error_constants.py [hunk 43]
```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python
```
Comment (seagate-sarang-sawant) [inline commons/error_constants.py:1]: Can you rename the file error_messages?

File: commons/error_constants.py (hunk 41 above)
Comment (seagate-sarang-sawant) [reply commons/error_constants.py:72]: I think @archanabpatil  is write. Duplicate key error should have a different text.  We can rename this variable as MALFORMED_XML.

File: commons/error_constants.py [hunk 44]
```diff
@@ -0,0 +1,104 @@
...
+                             "copy an object to itself without changing the object's metadata, "
+                             "storage class, website redirect location or encryption attributes.")
+CORTX_ERR_COPY_OBJ_METADATA = ("An error occurred (InvalidRequest) when calling the CopyObject "
+                               "operation: This copy request is illegal because it is trying to "
+                               "copy an object to itself without changing the object's metadata, "
+                               "storage class, website redirect location or encryption attributes.")
+RGW_HEAD_OBJ_ERR = ("An error occurred (404) when calling the HeadObject operation: Not Found")
+CORTX_HEAD_OBJ_ERR = ("An error occurred (404) when calling the HeadObject operation: Not Found")
+
+ACCOUNT_ERR = "attempted to create an account that already exists"
```
Comment (seagate-sarang-sawant) [inline commons/error_constants.py:57]: If these messages are S3 protocol specific only then write such in a comment. If they are prefixed by S3_ then it would be easily recognizable that a message text is s3 response error message.

File: tests/blackbox/test_aws_cli_s3api.py [hunk 45]
```diff
@@ -33,7 +33,7 @@
 from commons.utils import assert_utils
 from commons.utils import system_utils
 from commons.utils.s3_utils import assert_s3_err_msg
-from commons import constants as const
+from commons import error_constants as errconst
```
Comment (seagate-sarang-sawant) [inline tests/blackbox/test_aws_cli_s3api.py:36]: change references .

File: tests/s3/test_bucket_tagging.py [hunk 46]
```diff
@@ -170,7 +171,8 @@ def test_2434(self):
     def test_2435(self):
         """Create a tag whose key is up to 128 Unicode characters in length."""
         self.log.info(
-            "STARTED: Create a tag whose key is up to 128 Unicode characters in length")
+            "STARTED: Create a tag whose key is up to 128 Unicode "
```
Comment (seagate-sarang-sawant) [inline tests/s3/test_bucket_tagging.py:174]: Why a different style is followed for log messages. It seems different from rest of the comments.

**Review by seagate-sarang-sawant — CHANGES_REQUESTED** [review]

File: tests/blackbox/test_aws_cli_s3api.py (hunk 45 above)
Comment (seagate-sarang-sawant) [inline tests/blackbox/test_aws_cli_s3api.py:36]: The import needs a fix .

File: tests/blackbox/test_s3cmd.py [hunk 47]
```diff
@@ -39,7 +39,7 @@
 from commons.utils.assert_utils import assert_not_in
 from commons.utils.s3_utils import assert_s3_err_msg
 from commons.utils import system_utils
-from commons import constants as const
+from commons import error_constants as errconst
```
Comment (seagate-sarang-sawant) [inline tests/blackbox/test_s3cmd.py:42]: correct import with new file name.

**Review by seagate-sarang-sawant — CHANGES_REQUESTED** [review]

**Review by archanabpatil — APPROVED** [review]

File: tests/blackbox/test_s3cmd.py (hunk 47 above)
Comment (AshwiniS-Borse) [reply tests/blackbox/test_s3cmd.py]: done

**Review by apoorva-rao — APPROVED** [review]
Approving only to unblock @AshwiniS-Borse in her further tasks. Ashwini as discussed this needs to be verified with further test executions and any discrepancies in code formatting needs to be self-reviewed and fixed

**Review by seagate-sarang-sawant — APPROVED** [review]
